app/src/main/java/org/literacyapp/service/ScreenOnService.java

```diff
@@ -0,0 +1,27 @@
+package org.literacyapp.service;
+
+import android.app.Service;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.IBinder;
+import android.util.Log;
+
+import org.literacyapp.receiver.ScreenOnReceiver;
+
+public class ScreenOnService extends Service {
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        Log.i(getClass().getName(), "onStartCommand");
+
+        ScreenOnReceiver screenOnReceiver = new ScreenOnReceiver();
+        registerReceiver(screenOnReceiver, new IntentFilter("android.intent.action.SCREEN_ON"));
```

Comment:
So in `ScreenOnReceiver.java` we might need to check if the screen is in fact switched on, and only proceed if that's true.